ui-v2/src/components/ui/run-state-change-dialog/run-state-change-dialog.tsx

```diff
@@ -0,0 +1,205 @@
+
+	const onSubmit = async (values: RunStateFormValues) => {
+		if (isCurrentState) {
+			return;
+		}
+
+		setIsSubmitting(true);
+		try {
+			await onSubmitChange(values);
+			onOpenChange(false);
+		} catch {
+			// Error is handled/displayed by the onSubmitChange promise
+			// (typically via a toast in the mutation's onError callback)
+		} finally {
+			setIsSubmitting(false);
+		}
+	};
+
+	return (
+		<Dialog open={open} onOpenChange={onOpenChange}>
+			<DialogContent aria-describedby={undefined}>
+				<DialogHeader>
+					<DialogTitle>{title}</DialogTitle>
+				</DialogHeader>
+
+				<Form {...form}>
+					<form
+						onSubmit={(e) => void form.handleSubmit(onSubmit)(e)}
+						className="space-y-4"
+					>
+						<div className="mb-4">
+							<FormLabel className="mb-2 block">Current State</FormLabel>
+							<StateBadge type={currentState.type} name={currentState.name} />
+						</div>
+
+						<FormField
+							control={form.control}
+							name="state"
+							render={({ field }) => (
+								<FormItem className="w-full">
+									<FormLabel>Desired State</FormLabel>
+									<FormControl>
+										<Select {...field} onValueChange={field.onChange}>
+											<SelectTrigger
+												aria-label="select state"
+												className="w-full"
+											>
+												<SelectValue placeholder="Select state">
+													{field.value && (
+														<StateBadge
+															type={field.value}
+															name={RUN_STATES[field.value]}
+														/>
+													)}
+												</SelectValue>
+											</SelectTrigger>
+											<SelectContent className="w-full min-w-[300px]">
+												<SelectGroup>
+													{Object.keys(RUN_STATES).map((key) => (
+														<SelectItem
+															key={key}
+															value={key}
+															className="flex items-center"
+															disabled={key === currentState.type}
+														>
+															<StateBadge
+																type={key as components["schemas"]["StateType"]}
+																name={RUN_STATES[key as RunStates]}
+															/>
+														</SelectItem>
+													))}
+												</SelectGroup>
+											</SelectContent>
+										</Select>
+									</FormControl>
+									<FormMessage />
+								</FormItem>
+							)}
+						/>
+
+						<FormField
+							control={form.control}
+							name="message"
+							render={({ field }) => (
+								<FormItem>
+									<FormLabel>Reason (Optional)</FormLabel>
+									<FormControl>
+										<Textarea
+											{...field}
+											placeholder="State changed manually via UI"
+										/>
+									</FormControl>
+									<FormMessage />
+								</FormItem>
+							)}
+						/>
+
+						<div className="flex flex-col gap-2 pt-4">
+							{isCurrentState && (
+								<p className="text-sm text-muted-foreground">
+									Please select a different state than the current one to
+									continue
+								</p>
+							)}
```

Comment:
I think it'd be simpler to filter out the current state from the displayed options. That would eliminate the need for a form watch.

You could do something like this:
```ts
const stateOptions = useMemo(()=>(
    Object.entries(RUN_STATES).filter([k, _v] => currentState.type !== k)
), [currentState])
```